src/esp/sensor/VisualSensor.cpp

```diff
@@ -22,5 +26,31 @@ void VisualSensor::bindRenderTarget(gfx::RenderTarget::uptr&& tgt) {
   tgt_ = std::move(tgt);
 }
 
+void VisualSensor::setTransformationFromSpec() {
+  if (spec_ == nullptr) {
+    LOG(ERROR) << "Cannot initialize sensor. the specification is null.";
+    return;
+  }
+
+  node().resetTransformation();
+
+  node().translate(Magnum::Vector3(spec_->position));
+  node().rotateX(Magnum::Rad(spec_->orientation[0]));
+  node().rotateY(Magnum::Rad(spec_->orientation[1]));
+  node().rotateZ(Magnum::Rad(spec_->orientation[2]));
+}
+
+void VisualSensor::updateResolution(float x_res, float y_res) {
```

Comment:
resolution is `vec2i`. `i` means integer. 

Also we do not use _ for the variable by our C++ convention, we use Camel style. Just look around the other functions.

`width` and `height` is better than `xRes` and `yRes` 

CAREFUL!! in computer graphics, we usually w x h, w always comes before the h. HOWEVER here we follow the CV convention, namly h comes first, h x w.

A good design is:
`void VisualSensor::setResolution(int height, int width);`

Also provide an overload function with argument vec2i directly.